lightning/src/ln/channelmanager.rs

```diff
@@ -1432,7 +1463,7 @@ macro_rules! handle_chan_restoration_locked {
 		let chanmon_update_is_none = chanmon_update.is_none();
 		let counterparty_node_id = $channel_entry.get().get_counterparty_node_id();
 		let res = loop {
-			let forwards: Vec<(PendingHTLCInfo, u64)> = $pending_forwards; // Force type-checking to resolve
+			let forwards: Vec<(PendingHTLCInfo, u64, Option<u64>)> = $pending_forwards; // Force type-checking to resolve
```

Comment:
if i add it to PendingHTLCInfo then I wouldn't also have to thread through this extra Option<u64> in the tuple, right? 

It does seem to make sense to just put it in there.  it will still need to be an Option<u64> because they're serialized as part of pending forwards, right?

There's sort of another weirdness:

seems like in the Receive case, we are setting amt_to_forward to the 'prev/incoming amt' which ends up being fine because the logic differs for receives even though it's the same object.

i guess if i add this new field, would we want to update the receive logic as well to use it? though old PendingHtlcInfo's will still have the forward field set.  so we'd probably just want to store the incoming amount in both fields.. which is going to look pretty confusing.